pkgs/by-name/ha/hare/package.nix

```diff
@@ -146,9 +146,8 @@ stdenv.mkDerivation (finalAttrs: {
       --prefix PATH : ${lib.makeBinPath [binutils-unwrapped harec qbe]}
   '';
 
-  setupHook = ./setup-hook.sh;
-
   passthru = {
+    hook = callPackage ./hook.nix { hare = finalAttrs.finalPackage; };
```

Comment:
i expect passthru attributes don't get spliced, which could explain why cross compiling doesn't work (`hare.hook` would represent the derivation built for the host platform, regardless of which `nativeBuildInputs`/`buildInputs` set you reference it from).

i'm not sure the proper fix. possibly some amount of `lib.mkScope`, `lib.recurseIntoAttrs`, or making the `hare` package itself function as the setup hook (with some `dontHareConfigure` option to disable it). i'll poke around at how meson does this.

Reply:
@onemoresuza i think i fixed this to support cross compilation (i.e. `pkgsCross.aarch64-multiplatform.<pname>`), also in the process made it so that `hare` _is_ the setupHook (much like how `meson`, `cmake` work). tested it with `bonsai`, and also built but didn't run `haredo`.

https://github.com/onemoresuza/nixpkgs/compare/hare-hook...uninsane:nixpkgs:hare-hook-cross